NullPointerException on startup in IDEDevModeMain

### Describe the bug

Hello all,

**Context**
I recently upgrade from 2.9.2.Final to 2.13.3-Final and was unable to start my Quarkus application.
I use a custom class with @QuarkusMain and inner class that implements QuarkusApplication to start my Quarkus application.
My pom.xml contains several custom dependencies including 'pom' and 'jar' types.

**Issue**

Since the upgrade I get the following error:
```
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
     at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:58)
     at io.quarkus.runtime.Quarkus.launchFromIDE(Quarkus.java:99)
     at io.quarkus.runtime.Quarkus.run(Quarkus.java:86)
     at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
     at com.orange.sgg.SggApp.main(SggApp.java:22)
 Caused by: java.lang.reflect.InvocationTargetException
     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     at java.base/java.lang.reflect.Method.invoke(Method.java:566)
     at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:56)
     ... 4 more
 Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NullPointerException
     at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:96)
     ... 9 more
 Caused by: java.lang.RuntimeException: java.lang.NullPointerException
     at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:151)
     at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:103)
     at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:92)
     ... 9 more
 Caused by: java.lang.NullPointerException
     at io.quarkus.deployment.dev.IDEDevModeMain.toModule(IDEDevModeMain.java:102)
     at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:62)
     at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:1)
     at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:148)
     ... 11 more
```

The problem is located in:
```
private DevModeContext.ModuleInfo toModule(ResolvedDependency module) throws BootstrapGradleException {

        String classesDir = null;
        final Set<Path> sourceParents = new LinkedHashSet<>();
        final PathList.Builder srcPaths = PathList.builder();
        final ArtifactSources sources = module.getSources();
        for (SourceDir src : sources.getSourceDirs()) { // Fails because "sources" is null for a "pom" dependency
```

**Fix**

I was able to start my Quarkus application excluding the 'pom' type dependencies:
```
    for (ResolvedDependency project : DependenciesFilter.getReloadableModules(appModel)) {
        if("pom".equals(project.getType())) {
            continue;
        }
        final ModuleInfo module = toModule(project);
```

May be not the best way to fix it but it do the trick waiting for the official fix :)

Thanks for your great work


### Expected behavior

The Quarkus app start using @QuarkusMain even if the pom.xml contains 'pom' dependencies.

### Actual behavior

The Quarkus app does not start using @QuarkusMain if the pom.xml contains 'pom' dependencies.

### How to Reproduce?

1. Create a Quarkus app
2. add a 'pom' type dependency
3. add a "main" class using @QuarkusMain
4. start using the "main" class

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11.0.9.1" 2020-11-04

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.3-Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

### Additional information

_No response_

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/devmode/DependenciesFilter.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/devmode/DependenciesFilter.java
@@ -20,6 +20,9 @@ public class DependenciesFilter {
         final Map<ArtifactKey, WorkspaceDependencies> modules = new HashMap<>();
         StringBuilder nonReloadable = null;
         for (ResolvedDependency d : appModel.getDependencies()) {
+            if (!d.isJar()) {
+                continue;
+            }
             if (d.isReloadable()) {
                 modules.put(d.getKey(), new WorkspaceDependencies(d));
             } else if (d.isWorkspaceModule()) {

